src/OpenLoco/Vehicles/VehicleBogie.cpp

```diff
@@ -1,16 +1,145 @@
+#include "../Entities/EntityManager.h"
 #include "../Interop/Interop.hpp"
 #include "Vehicle.h"
 
 using namespace OpenLoco::Interop;
 
 namespace OpenLoco::Vehicles
 {
+    static loco_global<VehicleBogie*, 0x01136124> vehicleUpdate_frontBogie;
+    static loco_global<VehicleBogie*, 0x01136128> vehicleUpdate_backBogie;
+    static loco_global<bool, 0x01136237> vehicleUpdate_frontBogieHasMoved; // remainingDistance related?
+    static loco_global<bool, 0x01136238> vehicleUpdate_backBogieHasMoved;  // remainingDistance related?
+    static loco_global<int32_t, 0x0113612C> vehicleUpdate_var_113612C;     // Speed
+    static loco_global<uint32_t, 0x01136114> vehicleUpdate_var_1136114;
+    static loco_global<int32_t, 0x01136130> vehicleUpdate_var_1136130; // Speed
+    static loco_global<EntityId, 0x0113610E> vehicleUpdate_collisionCarComponent;
+
     // 0x004AA008
     bool VehicleBogie::update()
+    {
+        vehicleUpdate_backBogie = vehicleUpdate_frontBogie;
+        vehicleUpdate_frontBogie = this;
+
+        if (mode == TransportMode::air || mode == TransportMode::water)
+        {
+            return true;
+        }
+
+        const auto oldPos = position;
+        vehicleUpdate_var_1136114 = 0;
+        sub_4B15FF(vehicleUpdate_var_113612C);
+
+        const auto hasMoved = oldPos != position;
+        vehicleUpdate_backBogieHasMoved = vehicleUpdate_frontBogieHasMoved;
+        vehicleUpdate_frontBogieHasMoved = hasMoved;
+
+        const auto stash1136130 = vehicleUpdate_var_1136130;
+        if (var_5E != 0)
+        {
+            auto unk = var_5E;
+            if (unk > 32)
+            {
+                unk = 64 - unk;
+            }
+            vehicleUpdate_var_1136130 = 500 + unk * 320;
+        }
+
+        updateRoll();
+        vehicleUpdate_var_1136130 = stash1136130;
+        if (vehicleUpdate_var_1136114 & (1 << 1))
+        {
+            sub_4AA464();
+            return false;
+        }
+        else if (!(vehicleUpdate_var_1136114 & (1 << 2)))
+        {
+            return true;
+        }
+
+        sub_4AA464();
+        explodeComponent();
+        var_0C |= Flags0C::unk_5;
+        var_5A &= ~(1u << 31);
+        var_5A >>= 3;
+        var_5A |= (1u << 31);
```

Comment:
this pattern is used a lot in this method, can we move it out to another method?
its used on lines 63(here), 76, 107, 114, 121